### PR TITLE
Elg async cleanup2

### PR DESCRIPTION
**Pull request by bretambrose**
Allows for event loop cleanup to be safely invoked and resolved from an event loop thread.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: source/event_loop.c [hunk 1]
```diff
@@ -94,6 +95,51 @@ void aws_event_loop_group_clean_up(struct aws_event_loop_group *el_group) {
...
+    AWS_FATAL_ASSERT(aws_thread_init(&cleanup_thread, el_group->allocator) == AWS_OP_SUCCESS);
+
+    struct aws_thread_options thread_options;
+    AWS_ZERO_STRUCT(thread_options);
+
+    AWS_FATAL_ASSERT(
+        aws_thread_launch(&cleanup_thread, s_event_loop_destroy_async_thread_fn, data, &thread_options) ==
+        AWS_OP_SUCCESS);
+
+    aws_thread_clean_up(&cleanup_thread);
```
Comment (JonathanHenson) [inline source/event_loop.c]: We don’t need to join?

Comment (JonathanHenson) [reply source/event_loop.c]: Nevermind, I’m dumb, you intentionally detached so it would stay async

Comment (bretambrose) [reply source/event_loop.c]: clean_up = detach

File: include/aws/io/event_loop.h [hunk 2]
```diff
@@ -115,6 +115,8 @@ struct aws_event_loop_group {
     struct aws_atomic_var current_index;
 };
 
+typedef void(aws_event_loop_group_cleanup_complete_fn)(void *user_data);
```
Comment (graebm) [inline include/aws/io/event_loop.h]: we usually pass along the pointer to the event-loop-group

Comment (bretambrose) [reply include/aws/io/event_loop.h]: In cases where it's not safe (the loop is gone) I think it's up to the user to create some sort of association via user data if they need to associate specific callbacks with specific loops.

**Pull request merged**